server/backend/test/steps/send-notification.steps.ts

```diff
@@ -1,74 +1,71 @@
 import { INestApplication } from '@nestjs/common';
-import { Test } from '@nestjs/testing';
 import { defineFeature, loadFeature } from 'jest-cucumber';
 import * as request from 'supertest';
-import { getRepositoryToken } from '@mikro-orm/nestjs';
-import { CamerasModule } from '../../src/cameras/cameras.module';
 import { Camera } from '../../src/cameras/camera.entity';
 import { Notification } from '../../src/cameras/notification.entity';
+import {
+  buildTestStack,
+  TestStack,
+  TEST_STACK_TIMEOUT,
+} from '../helpers/test-stack';
+import { EntityManager, EntityRepository } from '@mikro-orm/core';
+import { MikroORM } from '@mikro-orm/core';
+import { v4 } from 'uuid';
+import { CamerasModule } from '../../src/cameras/cameras.module';
+import { Group } from '../../src/groups/group.entity';
+import { User } from '../../src/users/user.entity';
 
 const feature = loadFeature('test/features/send-notification.feature');
-const cameraA = new Camera('Cam-A', 'camera-A-token');
-const cameraB = new Camera('Cam-B', 'camera-B-token');
 
 defineFeature(feature, (test) => {
   let app: INestApplication;
-  let notificationRepository;
+  let testStack: TestStack;
+  let cameraRepository: EntityRepository<Camera>;
+  let em: EntityManager;
 
   beforeAll(async () => {
-    /* FIXME: typescript. just define the MockNotificationRepository properly, extend the EntityRepository class */
-    notificationRepository = {
-      notifications: [],
-      persist: (notification: Notification) => {
-        return;
-      },
-      findByCameraID(id: string): Notification[] {
-        return [];
-      },
-    };
-    const moduleRef = await Test.createTestingModule({
-      imports: [CamerasModule],
-    })
-      .overrideProvider(getRepositoryToken(Notification))
-      .useValue(notificationRepository)
-      .compile();
+    testStack = await buildTestStack({ imports: [CamerasModule] });
 
-    app = moduleRef.createNestApplication();
+    em = testStack.module.get<MikroORM>(MikroORM).em.fork();
```

Comment:
I'm a little confused what is going on here. What is `em`? Is a process being forked here?

Reply:
`em` is the global [Entity Manager](https://mikro-orm.io/docs/entity-manager), which handles persisting all entities to the database (I think). [`fork()`](https://mikro-orm.io/docs/identity-map#forking-entity-manager) creates an entity manager with a new [identity map and request context](https://mikro-orm.io/docs/identity-map), and honestly it's a leftover "hack" from when I was desperately trying to get MikroORM to play nice. Forking is required when actually using MikroORM and Nest does this automatically, but the global context can be re-enabled to make testing easier ([docs](https://mikro-orm.io/docs/identity-map#global-identity-map)). I actually have the global context enabled [here](https://github.com/tucker-moore/human-detector/blob/7a5abdbe384e07f7f86b171934232357ea920bb5/server/backend/test/helpers/test-stack.ts#L44) and the tests for the `list-camera-groups` feature work just fine without forking, so forking here is unnecessary. Hope that makes sense, sorry for the wall of text. MikroORM is wacky